tests/new_character_test.cpp

```diff
@@ -0,0 +1,173 @@
+#include "catch/catch.hpp"
+
+#include "game.h"
+#include "itype.h"
+#include "item.h"
+#include "player.h"
+#include "profession.h"
+#include "scenario.h"
+#include "mutation.h"
+#include "string_id.h"
+
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <unordered_map>
+
+static std::vector<std::string> next_subset( const std::vector<std::string> &set )
+{
+    // Doing it this way conveniently returns a vector containing solely set[foo] before
+    // it returns any other vectors with set[foo] in it
+    static unsigned bitset = 0;
+    std::vector<std::string> ret;
+
+    ++bitset;
+    // Check each bit position for a match
+    for( unsigned idx = 0; idx < set.size(); idx++ ) {
+        if( bitset & ( 1 << idx ) ) {
+            ret.push_back( set[idx] );
+        }
+    }
+    return ret;
+}
+
+static bool try_set_traits( const std::vector<std::string> &traits )
+{
+    g->u.empty_traits();
+    // TODO: This line accounts for well over half of the execution time of the test case
+    g->u.add_traits(); // mandatory prof/scen traits
+    for( const std::string &tr : traits ) {
+        if( g->scen->forbidden_traits( tr ) || g->u.has_conflicting_trait( tr ) ) {
+            return false;
+        } else if( !g->scen->traitquery( tr ) && !mutation_branch::get( tr ).startingtrait ) {
+            return false;
+        } else if( !g->u.has_trait( tr ) ) {
+            g->u.set_mutation( tr );
+        }
+    }
+    return true;
+}
+
+static player get_sanitized_player()
+{
+    // You'd think that this hp stuff would be in the c'tor...
+    player ret = player();
+    ret.recalc_hp();
+    for( int i = 0; i < num_hp_parts; i++ ) {
+        ret.hp_cur[i] = ret.hp_max[i];
+    }
+    // Set these insanely high so can_eat doesn't return TOO_FULL
+    ret.set_hunger( 10000 );
+    ret.set_thirst( 10000 );
+    return ret;
+}
+
+// TODO: Due to character::add_traits, mutations (ie traits unavailable in the generic scenario)
+// are commented out and not checked.
+// When they're not commented out, the test takes 317 minutes (not a typo!).
+// When they are commented out, the test takes 3 minutes.
+
+TEST_CASE( "starting_items" ) {
+    // Every starting trait that interferes with food/clothing
+    const std::vector<std::string> mutations = {
```

Comment:
Would be better if you generated that list rather than hardcoding it.
Not needed or anything, it's just that it would catch all the edge cases you could miss in manual handling.